### PR TITLE
Update Dialog to latest material-ui

### PR DESCRIPTION
**Pull request by dwkaminsky**
Updates the Dialog `sd_material_ui` component to the latest `material_ui`. Should not change any functionality of the component.

image

### CODE REVIEW

File: src/lib/components/Dialog.react.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                     onClose={this.closeDialog}
+                     actions={actions}
+                     className={className}
+                     actionsContainerClassName={actionsContainerClassName}
+                     bodyClassName={bodyClassName}
+                     contentClassName={contentClassName}
+                     overlayClassName={overlayClassName}
+                     paperClassName={paperClassName}
+                     titleClassName={titleClassName}>
+            <div style={{"margin": "14px"}}>
```
Comment (coralvanda) [inline src/lib/components/Dialog.react.js:144]: Let's give the user control over this.

File: src/lib/components/Dialog.react.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                        "margin-right": "-12px",}}
+              >
+                <CloseIcon/>
+              </IconButton>
+    } else {
+      this.browserSideClose = <div/>
+    }
+
+    return (
+      <div id={id} className="sd-dialog">
```
Comment (coralvanda) [inline src/lib/components/Dialog.react.js:128]: We should probably let the user control this className.

**Comment by coralvanda** [comment]
@dwkaminsky can you record a GIF of the dialog in action?

File: src/lib/components/Dialog.react.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        <CssBaseline/>
+        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
+          <MuiDialog aria-labelledby="simple-dialog-title"
+                     open={this.state.open}
+                     scroll="body"
+                     fullWidth={true}
+                     onClose={this.closeDialog}
+                     actions={actions}
+                     className={className}
+                     actionsContainerClassName={actionsContainerClassName}
```
[the remark this answers is not in the document]
Comment (dwkaminsky) [reply src/lib/components/Dialog.react.js:138]: Which line are you referring to here?

**Comment by dwkaminsky** [comment]
> @dwkaminsky can you record a GIF of the dialog in action?

captured

File: src/lib/components/Dialog.react.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (coralvanda) [reply src/lib/components/Dialog.react.js:157]: The `actions` prop.

[the remark this answers is not in the document]
Comment (dwkaminsky) [reply src/lib/components/Dialog.react.js:157]: It looks like I'm going to have to take that out, as well as all of the old classnames we had. Those are all no longer supported...

image

**Comment by dwkaminsky** [comment]
I also added a fix to an import issue in the Drawer component

File: src/lib/components/Dialog.react.js [hunk 4]
```diff
@@ -68,16 +55,10 @@ const defaultProps = {
...
-  bodyClassName: null,
-  contentClassName: null,
-  overlayClassName: null,
-  paperClassName: null,
-  titleClassName: null,
   autoScrollBodyContent: false,
   useBrowserSideClose: false,
   style: null,
-  scroll: null,
+  scroll: '',
```
Comment (coralvanda) [inline src/lib/components/Dialog.react.js:61]: This is better now that the user can change it, but let's keep the value you had from before as the default.

File: src/lib/components/Drawer.react.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // @flow
 
 import React, { Component } from 'react';
-import { Drawer as MuiDrawer } from '@material-ui/core/Drawer';
+import MuiDrawer from '@material-ui/core/Drawer';
```
Comment (coralvanda) [inline src/lib/components/Drawer.react.js:4]: Thank you.

File: src/lib/components/Dialog.react.js (hunk 4 above)
Comment (dwkaminsky) [reply src/lib/components/Dialog.react.js:61]: When I leave it as null, I get an error. That's why I changed it

File: src/lib/components/Drawer.react.js (hunk 5 above)
Comment (dwkaminsky) [reply src/lib/components/Drawer.react.js:4]: No problem, this took me forever to figure out for a different component

File: src/lib/components/Dialog.react.js (hunk 4 above)
Comment (coralvanda) [reply src/lib/components/Dialog.react.js:61]: Gotcha. Can we change it to "body" as the default instead?

Comment (dwkaminsky) [reply src/lib/components/Dialog.react.js:61]: Sure

**Pull request merged**